### ISSUE TITLE
Surfacing data from clustering to plot coloured scatter charts in Hart85

### ISSUE THREAD

**Issue by gjwo** [nilmtk/nilmtk#446]
I would like to be be able to see the individual clusters  produce by the Hart 85 algorithms associated with scatter charts see issue #444 for the kind of charts. The best way to do this would seem to be to use colour to show the individual clusters. I have found an example to do this for the clustering library used here: http://scikit-learn.org/stable/auto_examples/cluster/plot_mean_shift.html  My problem is this uses information from the MeanShift object that is not available outside of the call to  hart85_means_shift_cluster() this is the example cluster and plot code:
```python
ms = MeanShift(bandwidth=bandwidth, bin_seeding=True)
ms.fit(X)
labels = ms.labels_
cluster_centers = ms.cluster_centers_

labels_unique = np.unique(labels)
n_clusters_ = len(labels_unique)

print("number of estimated clusters : %d" % n_clusters_)

###############################################################################
# Plot result
import matplotlib.pyplot as plt
from itertools import cycle

plt.figure(1)
plt.clf()

colors = cycle('bgrcmykbgrcmykbgrcmykbgrcmyk')
for k, col in zip(range(n_clusters_), colors):
    my_members = labels == k
    cluster_center = cluster_centers[k]
    plt.plot(X[my_members, 0], X[my_members, 1], col + '.')
    plt.plot(cluster_center[0], cluster_center[1], 'o', markerfacecolor=col,
             markeredgecolor='k', markersize=14)
plt.title('Estimated number of clusters: %d' % n_clusters_)
plt.show()
```
This is the clustering code from nilmtk:
```python
def hart85_means_shift_cluster(pair_buffer_df, cols):


    from sklearn.cluster import MeanShift
    # Creating feature vector
    cluster_df = pd.DataFrame()
    power_types = [col[1] for col in cols]
    if 'active' in power_types:
        cluster_df['active'] = pd.Series(pair_buffer_df.apply(lambda row:
                                                                   ((np.fabs(row['T1 Active']) + np.fabs(row['T2 Active'])) / 2), axis=1), index=pair_buffer_df.index)
    if 'reactive' in power_types:
        cluster_df['reactive'] = pd.Series(pair_buffer_df.apply(lambda row:
                                                                     ((np.fabs(row['T1 Reactive']) + np.fabs(row['T2 Reactive'])) / 2), axis=1), index=pair_buffer_df.index)

    X = cluster_df.values.reshape((len(cluster_df.index), len(cols)))
    ms = MeanShift(bin_seeding=True)
    ms.fit(X)
    labels = ms.labels_
    cluster_centers = ms.cluster_centers_
    labels_unique = np.unique(labels)
    return pd.DataFrame(cluster_centers, columns=cols)
```
I think for the colouring to work the dataframe X, and the labels (or number of unique labels) needs to be available to a notebook. Could this information be stored in the hart85 class so that it could be accessible? Or is there a better solution? More generally in the future a set of predefined presentations of the data (mostly graphs) might be worth developing in the future to understand what the disaggregation algorithms are producing.
